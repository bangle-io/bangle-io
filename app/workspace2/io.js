import * as idb from 'idb-keyval';

export class FileOps {
  // file
  static async getFile(wsPath) {
    return idb.get(wsPath);
  }

  static async deleteFile(wsPath) {
    const file = await this.getFile(wsPath);
    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.del(wsPath);
  }

  static async renameFile(wsPath, newWsPath) {
    const file = await this.getFile(wsPath);

    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }

    const newFile = await this.getFile(newWsPath);

    if (newFile) {
      throw new FileError(`File ${newWsPath} already exists`);
    }

    await this.deleteFile(wsPath);
    return this.createFile(newWsPath, file);
  }

  static async createFile(wsPath, payload) {
    if (await this.getFile(wsPath)) {
      throw new FileError(`File ${wsPath} already exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async updateFile(wsPath, payload) {
    if (!(await this.getFile(wsPath))) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async listFiles(wsName) {
    const keys = (await idb.keys()) || [];

    return keys.filter((k) => k.startsWith(wsName + ':'));
  }
}

export class FileError extends Error {
  constructor(message) {
    super('FileError: ' + message);
    this.name = 'FileError';
  }
}
export class FilePermissionError extends Error {
  constructor(message) {
    super('FilePermissionError: ' + message);
    this.name = 'FilePermissionError';
  }
}
