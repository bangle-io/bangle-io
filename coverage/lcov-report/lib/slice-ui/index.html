
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for lib/slice-ui</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> lib/slice-ui</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">78.37% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>87/111</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">35.48% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>11/31</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">82.6% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>19/23</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.57% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>83/107</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="95.23" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 95%"></div><div class="cover-empty" style="width: 5%"></div></div>
	</td>
	<td data-value="95.23" class="pct high">95.23%</td>
	<td data-value="42" class="abs high">40/42</td>
	<td data-value="54.54" class="pct medium">54.54%</td>
	<td data-value="11" class="abs medium">6/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="40" class="abs high">38/40</td>
	</tr>

<tr>
	<td class="file medium" data-value="slice-ui-color-scheme.ts"><a href="slice-ui-color-scheme.ts.html">slice-ui-color-scheme.ts</a></td>
	<td data-value="59.09" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 59%"></div><div class="cover-empty" style="width: 41%"></div></div>
	</td>
	<td data-value="59.09" class="pct medium">59.09%</td>
	<td data-value="22" class="abs medium">13/22</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="slice-ui-widescreen.ts"><a href="slice-ui-widescreen.ts.html">slice-ui-widescreen.ts</a></td>
	<td data-value="72.34" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 72%"></div><div class="cover-empty" style="width: 28%"></div></div>
	</td>
	<td data-value="72.34" class="pct medium">72.34%</td>
	<td data-value="47" class="abs medium">34/47</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="12" class="abs low">4/12</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="71.73" class="pct medium">71.73%</td>
	<td data-value="46" class="abs medium">33/46</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-12-02T21:35:16.931Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    