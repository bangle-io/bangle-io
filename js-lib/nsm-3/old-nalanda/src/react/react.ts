import { useDebugValue, useRef, useState } from 'react';
import useSyncExternalStoreExports from 'use-sync-external-store/shim';

import type { Slice } from '../vanilla/slice';
import type { Store } from '../vanilla/store';
import type { AnySlice } from '../vanilla/types';

const { useSyncExternalStore } = useSyncExternalStoreExports;

type InferData<T> = T extends Slice<any, infer TData, any> ? TData : never;

interface ReactAdapter<TSnapshot> {
  subscribe: (onStoreChange: () => void) => () => void;
  getSnapshot: () => TSnapshot;
}

export function createUseTrackSliceHook<TAllSliceName extends string = any>(
  store: Store<TAllSliceName>,
) {
  function useSlice<TSlice extends AnySlice>(sl: TSlice): InferData<TSlice> {
    const ref = useRef<InferData<TSlice>>();

    const [adapter] = useState(() => {
      const adapter: ReactAdapter<InferData<TSlice>> = {
        subscribe: (onStoreChange) => {
          const sliceEffect = store.effect((effectStore) => {
            const selectedData = sl.track(effectStore);
            ref.current = selectedData;

            onStoreChange();

            queueMicrotask(() => {
              if (effectStore._runInstance?._addTrackedCount === 0) {
                console.warn(
                  `You forgot to destructure/access the tracked field from ${sl.name}. This will not track any changes!.`,
                );
              }
            });
          });

          return () => {
            store.unregisterEffect(sliceEffect);
          };
        },
        getSnapshot: () => {
          return ref.current ?? sl.get(store.state);
        },
      };

      return adapter;
    });

    const data = useSyncExternalStore(adapter.subscribe, adapter.getSnapshot);

    useDebugValue(data);

    return data;
  }

  return useSlice;
}
