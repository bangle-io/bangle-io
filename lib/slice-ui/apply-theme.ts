import type { ThemeType } from '@bangle.io/shared-types';

import { cssVars } from './css-vars';

export function applyTheme(theme: ThemeType) {
  console.debug('applying theme', theme);
  const element = document.documentElement;
  for (const cssVar of cssVars) {
    updateStyleHelper(theme, cssVar, element);
  }
}

function updateStyleHelper(
  theme: ThemeType,
  cssVar: string,
  element: HTMLElement,
) {
  element.style.setProperty(`--BV-${cssVar}`, `var(--BV-${theme}-${cssVar})`);
}
