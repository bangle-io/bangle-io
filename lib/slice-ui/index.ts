export {
  type UISliceState,
  initialUISliceState as initialState,
} from './constants';
export { nsmUISlice, uiEffects } from './nsm-ui-slice';
export * as nsmUI from './nsm-ui-slice';
