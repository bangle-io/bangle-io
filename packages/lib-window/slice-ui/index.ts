import { createKey } from '@nalanda/core';

import { sliceUIColorScheme } from './slice-ui-color-scheme';
import { sliceUIToast } from './slice-ui-toast';
import { initiallyWidescreen, sliceUIWidescreen } from './slice-ui-widescreen';

export {
  clearAllToast,
  clearToast,
  getToastEmitter,
  queueToast,
} from './slice-ui-toast';

const ONLY_ONE_ASIDE_THRESHOLD = 1024;

const key = createKey('slice-ui', [sliceUIWidescreen, sliceUIColorScheme]);

const showRightAsideField = key.field(false);

function toggleRightAside(show: boolean | undefined) {
  const tx = key.transaction();

  return tx.step((state) => {
    const { widescreen, screenWidth } = sliceUIWidescreen.get(state);
    const cur = showRightAsideField.get(state);

    if (!widescreen) {
      return state.apply(showRightAsideField.update(false));
    }

    // minimize the left aside when cramped
    if (screenWidth < ONLY_ONE_ASIDE_THRESHOLD) {
      state = state.apply(showLeftAsideField.update(false));
    }

    return state.apply(
      showRightAsideField.update(typeof show === 'boolean' ? show : !cur),
    );
  });
}

const showLeftAsideField = key.field(initiallyWidescreen ? true : false);

function toggleLeftAside(show: boolean | undefined) {
  const tx = key.transaction();

  return tx.step((state) => {
    const { widescreen, screenWidth } = sliceUIWidescreen.get(state);
    const currentShowLeftAside = showLeftAsideField.get(state);

    if (!widescreen) {
      return state.apply(showLeftAsideField.update(false));
    }

    // if we are showing the right aside, then we should hide it
    // since the screen is not wide enough
    if (screenWidth < ONLY_ONE_ASIDE_THRESHOLD) {
      state = state.apply(showRightAsideField.update(false));
    }

    return state.apply(
      showLeftAsideField.update(
        typeof show === 'boolean' ? show : !currentShowLeftAside,
      ),
    );
  });
}

const showActivitybarField = key.field(false);
function toggleActivitybar(show: boolean | undefined) {
  return showActivitybarField.update((cur) => {
    if (typeof show === 'boolean') {
      return show;
    }
    return !cur;
  });
}

const widescreenField = key.derive((state) => {
  return sliceUIWidescreen.getField(state, 'widescreen');
});

const screenWidthField = key.derive((state) => {
  return sliceUIWidescreen.getField(state, 'screenWidth');
});

const screenHeightField = key.derive((state) => {
  return sliceUIWidescreen.getField(state, 'screenHeight');
});

const colorSchemeField = key.derive((state) =>
  sliceUIColorScheme.getField(state, 'colorScheme'),
);

export const sliceUI = key.slice({
  colorScheme: colorSchemeField,
  screenHeight: screenHeightField,
  screenWidth: screenWidthField,
  showActivitybar: showActivitybarField,
  showLeftAside: showLeftAsideField,
  showRightAside: showRightAsideField,
  toggleActivitybar,
  toggleLeftAside,
  toggleRightAside,
  widescreen: widescreenField,
});

export const sliceUIAllSlices = [
  sliceUIWidescreen,
  sliceUIColorScheme,
  sliceUI,
  sliceUIToast,
];
