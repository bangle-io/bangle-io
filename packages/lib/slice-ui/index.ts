import { createKey } from '@nalanda/core';

import { sliceUIColorScheme } from './slice-ui-color-scheme';
import { sliceUIWidescreen } from './slice-ui-widescreen';

const key = createKey('slice-ui', [sliceUIWidescreen, sliceUIColorScheme]);

const showRightAsideField = key.field(false);
function toggleRightAside(show: boolean | undefined) {
  return showRightAsideField.update((cur) => {
    if (typeof show === 'boolean') {
      return show;
    }
    return !cur;
  });
}

const showLeftAsideField = key.field(false);
function toggleLeftAside(show: boolean | undefined) {
  return showLeftAsideField.update((cur) => {
    if (typeof show === 'boolean') {
      return show;
    }
    return !cur;
  });
}

const showActivitybarField = key.field(false);
function toggleActivitybar(show: boolean | undefined) {
  return showActivitybarField.update((cur) => {
    if (typeof show === 'boolean') {
      return show;
    }
    return !cur;
  });
}

const widescreenField = key.derive((state) => {
  return sliceUIWidescreen.getField(state, 'widescreen');
});

const screenWidthField = key.derive((state) => {
  return sliceUIWidescreen.getField(state, 'screenWidth');
});

const screenHeightField = key.derive((state) => {
  return sliceUIWidescreen.getField(state, 'screenHeight');
});

const colorSchemeField = key.derive((state) =>
  sliceUIColorScheme.getField(state, 'colorScheme'),
);

export const sliceUI = key.slice({
  colorScheme: colorSchemeField,
  screenHeight: screenHeightField,
  screenWidth: screenWidthField,
  showActivitybar: showActivitybarField,
  showLeftAside: showLeftAsideField,
  showRightAside: showRightAsideField,
  toggleActivitybar,
  toggleLeftAside,
  toggleRightAside,
  widescreen: widescreenField,
});

export const sliceUIAllSlices = [
  sliceUIWidescreen,
  sliceUIColorScheme,
  sliceUI,
];
