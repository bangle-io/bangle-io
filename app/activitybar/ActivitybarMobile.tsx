import React from 'react';

import { useBangleStoreContext, vars } from '@bangle.io/api';
import { CorePalette } from '@bangle.io/constants';
import { useExtensionRegistryContext } from '@bangle.io/extension-registry';
import {
  toggleEditing,
  useEditorManagerContext,
} from '@bangle.io/slice-editor-manager';
import { togglePaletteType, useUIManagerContext } from '@bangle.io/slice-ui';
import {
  goToWsNameRoute,
  useWorkspaceContext,
} from '@bangle.io/slice-workspace';
import { Button, ChevronLeftIcon } from '@bangle.io/ui-components';
import { resolvePath } from '@bangle.io/ws-path';

import { ActivitybarOptionsDropdown } from './ActivitybarOptionsDropdown';

export function ActivitybarMobile() {
  const extensionRegistry = useExtensionRegistryContext();
  const bangleStore = useBangleStoreContext();
  const { wsName, openedWsPaths } = useWorkspaceContext();
  const { primaryWsPath } = openedWsPaths;

  const { editingAllowed: showDone } = useEditorManagerContext();
  const { sidebar: activeSidebar } = useUIManagerContext();

  return (
    <ActivitybarMobileDumb
      bangleStore={bangleStore}
      primaryWsPath={primaryWsPath}
      wsName={wsName}
      showDone={showDone}
      extensionRegistry={extensionRegistry}
      activeSidebar={activeSidebar}
    />
  );
}

export function ActivitybarMobileDumb({
  bangleStore,
  primaryWsPath,
  wsName,
  showDone,
  extensionRegistry,
  activeSidebar,
}: {
  bangleStore: ReturnType<typeof useBangleStoreContext>;
  primaryWsPath: string | undefined;
  wsName: string | undefined;
  showDone: boolean;
  extensionRegistry: ReturnType<typeof useExtensionRegistryContext>;
  activeSidebar: ReturnType<typeof useUIManagerContext>['sidebar'];
}) {
  const operationKeybindings =
    extensionRegistry.getSerialOperationKeybindingMapping();

  const sidebarItems = extensionRegistry.getSidebars().filter((r) => {
    return r.activitybarIconShow
      ? r.activitybarIconShow(wsName, bangleStore.state)
      : true;
  });

  return (
    <>
      <div
        data-testid="app-activitybar_activitybar-mobile"
        style={{
          backgroundColor: vars.misc.activitybarBg,
        }}
        className="flex flex-row px-2 items-center w-full select-none"
      >
        <div className="flex flex-row items-center flex-none">
          {primaryWsPath ? (
            <Button
              variant="transparent"
              ariaLabel="Go home"
              size="md"
              leftIcon={<ChevronLeftIcon />}
              onPress={() => {
                goToWsNameRoute(resolvePath(primaryWsPath).wsName, {
                  reopenPreviousEditors: false,
                  replace: false,
                })(bangleStore.state, bangleStore.dispatch);
              }}
            />
          ) : null}
        </div>

        <div
          className="flex flex-row items-center shrink overflow-hidden"
          role="button"
        >
          <Button
            variant="transparent"
            ariaLabel="files palette"
            size="md"
            text={
              primaryWsPath
                ? resolvePath(primaryWsPath).fileNameWithoutExt
                : wsName || 'bangle-io'
            }
            onPress={() => {
              togglePaletteType(CorePalette.Notes)(
                bangleStore.state,
                bangleStore.dispatch,
              );
            }}
          />
        </div>
        <div className="flex flex-1"></div>
        <div className="flex flex-row items-center flex-none">
          {primaryWsPath && (
            <div className="mr-2">
              {showDone ? (
                <Button
                  variant="solid"
                  tone="promote"
                  ariaLabel="done editing"
                  className="capitalize"
                  size="sm"
                  text="Done"
                  onPress={() => {
                    toggleEditing({
                      focusOrBlur: true,
                      editingAllowed: false,
                    })(bangleStore.state, bangleStore.dispatch);
                  }}
                />
              ) : (
                <Button
                  variant="solid"
                  ariaLabel={'edit'}
                  className="capitalize"
                  size="sm"
                  text="edit"
                  onPress={() => {
                    toggleEditing({
                      focusOrBlur: true,
                      editingAllowed: true,
                    })(bangleStore.state, bangleStore.dispatch);
                  }}
                />
              )}
            </div>
          )}

          <div className="mr-2">
            <ActivitybarOptionsDropdown
              operationKeybindings={operationKeybindings}
              sidebarItems={sidebarItems}
              activeSidebar={activeSidebar || undefined}
              widescreen={false}
            />
          </div>
        </div>
      </div>
    </>
  );
}
