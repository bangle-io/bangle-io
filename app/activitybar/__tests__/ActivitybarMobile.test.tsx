/**
 * @jest-environment @bangle.io/jsdom-env
 */
import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';

import { togglePaletteType } from '@bangle.io/slice-ui';
import {
  createBasicTestStore,
  setupMockWorkspaceWithNotes,
  TestStoreProvider,
} from '@bangle.io/test-utils';

import { ActivitybarMobile } from '../ActivitybarMobile';

jest.mock('@bangle.io/slice-ui', () => {
  const otherThings = jest.requireActual('@bangle.io/slice-ui');

  return {
    ...otherThings,
    togglePaletteType: jest.fn(() => jest.fn(() => {})),
  };
});

let togglePaletteTypeMock = togglePaletteType as jest.MockedFunction<
  typeof togglePaletteType
>;

test('renders mobile view', async () => {
  const { store } = createBasicTestStore({
    extensions: [],
    useEditorManagerSlice: true,
    useUISlice: true,
  });

  await setupMockWorkspaceWithNotes(store);

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <ActivitybarMobile></ActivitybarMobile>
      </TestStoreProvider>
    </div>,
  );

  expect(result.container).toMatchSnapshot();
});

test('renders primaryWsPath', async () => {
  const { store } = createBasicTestStore({
    extensions: [],
    useEditorManagerSlice: true,
    useUISlice: true,
  });

  await setupMockWorkspaceWithNotes(store, 'my-thing', [
    ['my-thing:wow1.md', 'wow1'],
    // last created file will be the primaryWsPath
    ['my-thing:wow2.md', 'wow2'],
  ]);

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <ActivitybarMobile />
      </TestStoreProvider>
    </div>,
  );

  expect(result.container.innerHTML).toContain('wow2');
});

test('dispatches operation', async () => {
  const { store } = createBasicTestStore({
    extensions: [],
    useEditorManagerSlice: true,
    useUISlice: true,
  });

  await setupMockWorkspaceWithNotes(store, 'my-thing', [
    ['my-thing:wow1.md', 'wow1'],
    // last created file will be the primaryWsPath
    ['my-thing:wow2.md', 'wow2'],
  ]);

  let result = render(
    <div>
      <TestStoreProvider bangleStore={store} bangleStoreChanged={0}>
        <ActivitybarMobile />
      </TestStoreProvider>
    </div>,
  );
  act(() => {
    fireEvent.click(result.getByLabelText('files palette'));
  });

  expect(togglePaletteTypeMock).toBeCalledTimes(1);
  expect(togglePaletteTypeMock).nthCalledWith(
    1,
    'bangle-io-core-palettes/notes',
  );
});
