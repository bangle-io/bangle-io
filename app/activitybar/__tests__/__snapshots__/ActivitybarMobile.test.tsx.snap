// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mobile view 1`] = `
<div>
  <div>
    <div
      class="flex flex-row px-2 items-center w-full select-none"
      data-testid="app-activitybar_activitybar-mobile"
    >
      <div
        class="flex flex-row items-center flex-none"
      >
        <button
          aria-label="Go home"
          class="text-base h-9 smallscreen:h-10 min-w-10 px-3  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
          style="background-color: transparent;"
          type="button"
        >
          <span
            class="flex flex-grow-1 overflow-hidden "
            style="justify-content: center;"
          >
            <span>
              <svg
                class="w-6 h-6  "
                fill="currentColor"
                viewBox="0 0 18 18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#ff13dc"
                  height="18"
                  opacity="0"
                  width="18"
                />
                <path
                  d="M6,9a.994.994,0,0,0,.2925.7045l3.9915,3.99a1,1,0,1,0,1.4355-1.386l-.0245-.0245L8.4095,9l3.286-3.285A1,1,0,0,0,10.309,4.28l-.0245.0245L6.293,8.2945A.994.994,0,0,0,6,9Z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <div
        class="flex flex-row items-center shrink overflow-hidden"
        role="button"
      >
        <button
          aria-label="files palette"
          class="text-base h-9 smallscreen:h-10 min-w-10 px-3  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
          style="background-color: transparent;"
          type="button"
        >
          <span
            class="flex flex-grow-1 overflow-hidden "
            style="justify-content: center;"
          >
            <span
              class="text-ellipsis overflow-hidden "
            >
              two
            </span>
          </span>
        </button>
      </div>
      <div
        class="flex flex-1"
      />
      <div
        class="flex flex-row items-center flex-none"
      >
        <div
          class="mr-2"
        >
          <button
            aria-label="done editing"
            class="capitalize text-sm font-600 h-8 min-w-8 px-2  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
            type="button"
          >
            <span
              class="flex flex-grow-1 overflow-hidden "
              style="justify-content: center;"
            >
              <span
                class="text-ellipsis overflow-hidden "
              >
                Done
              </span>
            </span>
          </button>
        </div>
        <div
          class="mr-2"
        >
          <button
            aria-label="options menu"
            aria-pressed="false"
            class="text-sm font-600 h-8 min-w-8 px-2  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
            style="background-color: transparent;"
            type="button"
          >
            <span
              class="flex flex-grow-1 overflow-hidden "
              style="justify-content: center;"
            >
              <span>
                <svg
                  class="w-5 h-5  "
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
